packages/shared/lib/sdk/dataValidation.ts

```diff
@@ -1,38 +1,55 @@
-import type { ErrorObject } from 'ajv';
+import type { ErrorObject, ValidateFunction } from 'ajv';
 import { Ajv } from 'ajv';
 import addFormats from 'ajv-formats';
 import type { JSONSchema7 } from 'json-schema';
 
 export interface ValidateProps {
+    version: string;
     input: unknown;
     modelName: string | undefined;
     jsonSchema: JSONSchema7 | undefined | null;
 }
 
-export function validateData({ input, modelName, jsonSchema }: ValidateProps): true | (ErrorObject | Error)[] {
+let ajv: Ajv;
+const cache = new Map<string, ValidateFunction<any>>();
```

Comment:
we don't limit the size of the cache. I guess it is not so much of a problem at the moment since it is one per customer/runner and also limited by the number of model/version. Might become a problem if we start sharing runners in the furure